airflow/contrib/operators/aws_athena_operator.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.contrib.hooks.aws_athena_hook import AWSAthenaHook
+
+
+class AWSAthenaOperator(BaseOperator):
+    """
+    Airflow operator to run presto queries on athena.
+
+    :param query: Presto to be run on athena. (templated)
+    :type query: str
+    :param database: Database to select. (templated)
+    :type database: str
+    :param output_location: s3 path to write the query results into. (templated)
+    :type output_location: str
+    :param aws_conn_id: aws connection to use.
+    :type aws_conn_id: str
+    """
+
+    ui_color = '#44b5e2'
+    template_fields = ('query', 'database', 'output_location')
+
+    @apply_defaults
+    def __init__(self, query, database, output_location, aws_conn_id='aws_default', *args, **kwargs):
+        super(AWSAthenaOperator, self).__init__(*args, **kwargs)
+        self.query = query
+        self.database = database
+        self.output_location = output_location
+        self.aws_conn_id = aws_conn_id
+        self.client_request_token = kwargs.get('client_request_token')
+        self.query_execution_context = kwargs.get('query_execution_context') or {}
+        self.result_configuration = kwargs.get('result_configuration') or {}
+        self.query_execution_context['Database'] = self.database
```

Comment:
I think this is too early - and will happen before the templates are applied to self.database etc - templates are applied on after the constructor.

Reply:
I am not aware of that. Can you tell me where is the correct place to initialize above variable for the operator.

Reply:
Templates are not rendered until just before `execute()` is called, so these two things probably just need to be delayed until then.

Another option might be to just declare `query_execution_context` as templated (then it will look for/resolve templates in the values of the dict) but I think this interface you have is more direct and likely more friendly.